main.go

```diff
@@ -29,140 +31,155 @@ import (
 	genericapiserver "k8s.io/apiserver/pkg/server"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/component-base/logs"
-	"k8s.io/klog/v2"
 	"sigs.k8s.io/custom-metrics-apiserver/pkg/apiserver"
 	basecmd "sigs.k8s.io/custom-metrics-apiserver/pkg/cmd"
-	cm_provider "sigs.k8s.io/custom-metrics-apiserver/pkg/provider"
+
+	"go.elastic.co/apm"
 
 	generatedopenapi "github.com/elastic/elasticsearch-k8s-metrics-adapter/generated/openapi"
 	"github.com/elastic/elasticsearch-k8s-metrics-adapter/pkg/client"
 	"github.com/elastic/elasticsearch-k8s-metrics-adapter/pkg/client/custom_api"
 	"github.com/elastic/elasticsearch-k8s-metrics-adapter/pkg/client/elasticsearch"
 	"github.com/elastic/elasticsearch-k8s-metrics-adapter/pkg/config"
+	"github.com/elastic/elasticsearch-k8s-metrics-adapter/pkg/log"
 	"github.com/elastic/elasticsearch-k8s-metrics-adapter/pkg/monitoring"
 	"github.com/elastic/elasticsearch-k8s-metrics-adapter/pkg/provider"
 	"github.com/elastic/elasticsearch-k8s-metrics-adapter/pkg/registry"
 	"github.com/elastic/elasticsearch-k8s-metrics-adapter/pkg/scheduler"
 	"github.com/elastic/elasticsearch-k8s-metrics-adapter/pkg/tracing"
-	"go.elastic.co/apm"
 )
 
+const (
+	serviceType         = "elasticsearch-k8s-metrics-adapter"
+	serviceVersion      = "0.0.0"
+	defaultLogVerbosity = 0
+
+	elastisearchMetricServerType = "elasticsearch"
+	customMetricServerType       = "custom"
+)
+
+var (
+	logger logr.Logger
+)
+
+func main() {
+	log.Configure(defaultLogVerbosity, serviceType, serviceVersion)
+	logger = log.ForPackage("main")
+
+	logs.InitLogs()
+	defer logs.FlushLogs()
```

Comment:
We could probably pull this into `log.Configure`? 
Something like: 
```
defer log.Configure(defaultLogVerbosity, serviceType, serviceVersion)
```
where `log.Configure` returns a function for defer

Reply:
I can move it but it would still be:
```go
flushLog := log.Configure(defaultLogVerbosity, serviceType, serviceVersion)
defer flushLog()
```
Otherwise `log.Configure()` is not called to initialize the loggers, because it is precisely deferred?